contracts/exchange/test/core.ts

```diff
@@ -69,7 +70,7 @@ describe('Exchange core', () => {
     let exchange: ExchangeContract;
     let erc20Proxy: ERC20ProxyContract;
     let erc721Proxy: ERC721ProxyContract;
-    let erc1155Proxy: ERC721ProxyContract;
+    let erc1155Proxy: ERC1155ProxyContract;
```

Comment:
An old oversight that was untested? (I saw there were multiple instances of this.)  Is it tested now?

Reply:
yeah, these were weird. Maybe @hysz knows something about why they were written this way?

What's happening in this PR is that `*Contract` classes in `abi-gen-wrappers` are now typed (via the SubscriptionManager) to the type of events they have. Previously, the typescript compiler treated `ERC721ProxyContract` and `ERC1155ProxyContract` as interchangeable because they had identical interfaces. With `SubscriptionManager<ContractSpecificEventArgs, ContractSpecificEvent>`, they stop being interchangeable. Without this fix, the compiler complains about these lines later in the file:

```
erc1155ProxyWrapper = new ERC1155ProxyWrapper(provider, usedAddresses, owner);
erc1155Proxy = await erc1155ProxyWrapper.deployProxyAsync();
```

Reply:
Good find! Definitely a typo. I'm surprised this compiled/worked at all ??

This variable is later assigned as follows:
```
erc1155Proxy = await erc1155ProxyWrapper.deployProxyAsync();
```

Where, in `ERC1155ProxyWrapper`:
```
/**
     * @dev Deploys the ERC1155 proxy
     * @return Deployed ERC1155 proxy contract instance
     */
    public async deployProxyAsync(): Promise<ERC1155ProxyContract> {
        this._proxyContract = await ERC1155ProxyContract.deployFrom0xArtifactAsync(
            artifacts.ERC1155Proxy,
            this._provider,
            txDefaults,
        );
        this._proxyIdIfExists = await this._proxyContract.getProxyId.callAsync();
        return this._proxyContract;
    }
```

Reply:
Yeah, the compiler is happy if the variable instance has all the required fields of the type. I gather this is so that it can work with random js objects. Yikes.